app/src/main/java/de/danoeh/antennapod/fragment/SearchFragment.java

```diff
@@ -241,14 +260,56 @@ public boolean onMenuItemActionCollapse(MenuItem item) {
 
     @Override
     public boolean onContextItemSelected(@NonNull MenuItem item) {
+
+        int itemId = item.getItemId();
         FeedItem selectedItem = adapter.getLongPressedItem();
+        Feed selectedFeedItem  = adapterFeeds.getLongPressedItem();
+
+        if (selectedFeedItem != null) {
+            if (itemId == R.id.rename_folder_item) {
```

Comment:
We already have the same statement in another class somewhere. In order to avoid code duplication, please extract them to a new class. Something like `FeedItemMenuHandler.onMenuItemClicked`, just for feeds. Then newly added new menu items are always consistent between screens.